datafusion/expr/src/columnar_value.rs

```diff
@@ -183,6 +185,43 @@ impl ColumnarValue {
         Ok(args)
     }
 
+    fn _format_interval_string_value(&self, str_val: &String) -> Option<String> {
+        let start_idx = if str_val.starts_with('-') { 1 } else { 0 };
+
+        // only incase of simple input that all char is a number.
+        if str_val[start_idx..].chars().all(|c| c.is_ascii_digit()) {
+            return Some(format!("{} second", str_val));
+        }
+        None
+    }
+
+    fn _kernels_cast_with_option(
```

Comment:
maybe useful to add a docstring here explaining why this method is necessary / what it does.